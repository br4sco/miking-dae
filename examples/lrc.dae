variables
_u_R_1, _u_L_1, _u_C_1, _i_R_1, _i_L_1, _i_C_1, _y_1, _u_1, _t_1, _i_1 : Float
init
_t_1 = 0.
equations
_t_1' = 1.;
_u_1 = sin(_t_1);
_u_R_1 = 2. * _i_R_1;
_i_C_1 = 3. * _u_C_1';
_u_L_1 = 4. * _i_L_1';
_u_C_1 = 1. * _u_L_1;
_u_R_1 = (((0. - 1.) * 0. - 1. * _u_L_1) - (0. - 1.) * _u_1);
_i_1 = (0. - 1.) * _i_R_1;
_i_L_1 = (1. * _i_R_1 - 1. * _i_C_1);
_y_1 = 1. * _i_R_1
output

{_y_1}

