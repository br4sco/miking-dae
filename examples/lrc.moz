include GRAPHSIMVISUAL

def y, u, t: Real

def R = 2.
def C = 3.
def L = 4.

def n1, n2: Node

def RLC(y: Current, u: Voltage) -> Model = {
  n1 -- Resistor(R) -- CurrentSensor(y) -- (Capacitor(C) || Inductor(L)) -- n2;
  n1 -- VoltageSource(u) -- n2
}

def model = {
  Clock(0., t);
  u = sin(t);
  RLC(y, u);
  probe y
}

def _ = print (export2peadae (elaborate model) ++ "\n")
