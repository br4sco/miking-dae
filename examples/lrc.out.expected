0.461630416478
